Gems/Atom/Utils/Code/Tests/PngFileTests.cpp

```diff
@@ -0,0 +1,327 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Atom/Utils/PngFile.h>
+#include <AzCore/UnitTest/TestTypes.h>
+#include <AzCore/Math/Color.h>
+#include <AzCore/std/containers/array.h>
+#include <AzCore/IO/FileIO.h>
+#include <AzFramework/IO/LocalFileIO.h>
+
+namespace UnitTest
+{
+    using namespace AZ::Utils;
+
+    class PngFileTests
+        : public AllocatorsFixture
+    {
+    protected:
+        AZStd::string m_testImageFolder;
+        AZStd::string m_tempPngFilePath;
+        AZStd::vector<uint8_t> m_primaryColors3x1;
+        AZStd::unique_ptr<AZ::IO::FileIOBase> m_localFileIO;
+
+        void SetUp() override
+        {
+            AllocatorsFixture::SetUp();
+
+            m_testImageFolder = AZ::Test::GetEngineRootPath() + "/Gems/Atom/Utils/Code/Tests/PngTestImages/";
+            m_tempPngFilePath = m_testImageFolder + "temp.png";
```

Comment:
AZ::IO::Path should be used for Path manipulation instead of string manipulation.
It deals with trailing path slashes and provides an abstraction for properly dealing with paths.
```suggestion
            m_testImageFolder = AZ::IO::Path(AZ::Test::GetEngineRootPath()) / "Gems" / "Atom" / "Utils" / "Code" / "Tests" / "PngTestImages";
            m_tempPngFilePath = m_testImageFolder / "temp.png";
```